inc/Engine/Admin/Settings/Settings.php

```diff
@@ -693,7 +693,17 @@ private function check_global_exclusion( $field ) {
 	private function sanitize_cdn_cnames( array $cnames ) {
 		$cnames = array_map(
 			function( $cname ) {
-				return filter_var( $cname, FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME );
+				if ( empty( $cname ) ) {
+					return false;
+				}
+
+				$cname_parts = get_rocket_parse_url( rocket_add_url_protocol( $cname ) );
+
+				if( false === filter_var( $cname_parts['host'], FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME ) ) {
+					return false;
+				}
+
+				return $cname_parts['scheme'] . '://' . $cname_parts['host'];
```

Comment:
We also need to include the path if it exists, some CDN URL use a path value

Reply:
Good point, Done